runtime/vm/BufferWriter.hpp

```diff
@@ -333,6 +337,11 @@ class VM_BufferWriter {
 		writeU32(newVal);
 	}
 
+	void writeBoolean(BOOLEAN val)
+	{
+		writeU8((TRUE == val) ? 1 : 0);
```

Comment:
A boolean should only be tested directly or compared with false:
```c
		writeU8(val ? 1 : 0);
```
(comparing with true is an anti-pattern)